### DISCUSSION TITLE
JLBPCB assembly

### DISCUSSION THREAD

**Discussion by rawsp33d** [JellyTitan/Sofle-Pico#10] [Q&A] [answered]
I just ordered a batch of boards from JLB, and after doing so I remembered that they have their PCBA option where they assemble parts of your build based on your library of components and parts you buy from them. I've never done this before and do not have any experience doing so, but it would solve some problems I had. For example I could not find the LEDs _anywhere_ other than Aliexpress, and they apparently take an entire month to deliver anything. Thus I decided to just forego LEDs entirely for now.

Is the sofle-pico friendly with JLB's PCBA workflow? If so could somebody give me instructions or advice or anything on doing this? In particular the "economic" option only lets you have one side assembled, but since the boards are intentionally all the same I am kind of afraid I would get 5 left-handed or 5 right-handed keyboards :sweat_smile:.

Any insight you can provide for using their PCBA process for this board would be much appreciated!

**Comment by JellyTitan** [comment, marked as answer]
Great question! I've used PCBA on this board with the level shifters and the diodes. The diodes can be installed on either side, and the level shifter only on one side, so a single PCBA order with those works for both hands.

I have not tried it with the LEDs and Sockets, because those are considered 'special' parts, and there's an additional $25 charge for those. 

If you did want to use PCBA for those parts, you'd have to place a separate order for right and left hands, as well as modify the KiCad files. The current LED & hot swap sockets are set to be on the backside, so you'd need to add ghost footprints for PCBA on the opposite side. The Sofle Choc has those ghost footprints for the LED's if you want to see an example.

I've generated PCBA BOM/PNP files for earlier dev iterations of the board - but I stopped bothering with it because JLC started charging a shipping fee for the PCBA as well as the boards, which essentially doubled the cost of the shipping.

Eventually I'd like to get these ghost footprints setup, but right now a stable release and build guide are the priority.

@rosshadden please let me know if you have any suggestions as you build out these boards - I'd love input on the build guide!
